### ISSUE TITLE
Feature Request: Support for polar graphs

### ISSUE THREAD

**Issue by tedzards509** [Sjoerd1993/Graphs#715]
This should be low priority or be made an entirely separate app I think. Feel free to close this issue if you think so too.

I have not yet stumbled over a use for complex graphs, but they are supported by matplotlib: https://matplotlib.org/stable/gallery/pie_and_polar_charts/polar_demo.html

Also this could pave the way to implementing complex functions using Damian Coloring (similar to https://notebook.community/empet/Math/DomainColoring). This is not implemented directly in matplotlib.

**Comment by Sjoerd1993** [comment]
I will look into this. It may require some thinking design-wise, and some experimentation with how much we'd have to mess with the current implementation to make this possible. On the plus-side, the imported data is basically the same format (two arrays), so it might be doable to combine this relatively smoothly.

Haven't looked too much into detail into the implementation (apart from the matplotlib link you posted), but depending on how well this fits with the current code-base, then this might be reasonable. But e.g. error bars have a higher priority at the moment.

**Comment by Sjoerd1993** [comment]
I just looked into this superficially. But just to share my findings, it is possible to jam polar support into Graphs. There's a few challenges that need to be overcome:
- Support for multiple axes does not work, this does not make as much sense for polar coordinates either. But lots in the canvas logic works with multiple axes. This should thus require quite a bit of refactoring, perhaps it would make sense to split up the canvas class. One base class. And then two classes that inherit from this, one for line plots, and one for polar plots. This could make sense if we were to support a few more types as well
- Some other features do not work either. For instance the x-scale cannot be set in polar plots. This would thus also mean that those should be hidden for polar plots. Thus also adding a bit more work for that
- There's a few UI challenges. What happens if a user has data on multiple axes, and then switches to a polar plot for instance? Probably the answer here is to force all data to the single axes that is supported for polar plots, and then give a toast telling the user about this. Another one is that the "Highlight" tool won't work, so we should probably deactivate it if a polar plot is active (maybe the same is true for some operations such as combine and cut).

Anyway, here's what it looks like if I just jam it in without any changes:

![image](https://github.com/Sjoerd1993/Graphs/assets/68477016/239790be-eae1-4554-b8fb-a0b936b5235e)


The weird labels are due to the logic with multiple axes that we assume right now. The polar plots respect the theming, so that's nice. In general this should be done in multiple steps. I think it can be worth to think about adding support for other plot types in the future (I'd like to be able to do bar charts for instance). So the first step would be a refactor of `Canvas` so that we can make that more modular and support other types than line/scatter plots. Then we can think of adding support for other types as well. 

So in short, it will be a process. And I'm not directly expecting to be able to implement polar plots specifically with next release. But I think it would be nice to slowly work towards adding other plot types as well, including polar plots.
